### PR TITLE
nit-picking

### PR DESCRIPTION
**Pull request by athiruma**
## Type of change
Note: Fill **x** in []
- [ ] bug
- [x] enhancement
- [ ] documentation
- [ ] dependencies

## Description
pre-commit, scans code for secrets before they are committed and blocks the commit if any are detected.

### CODE REVIEW

**Comment by ebattat** [comment]
Pls add detailed description.
What is the purpose of this change ?

**Comment by ebattat** [comment]
Is it scan all the PR changes ?

File: .github/workflows/Build.yml [hunk 1]
```diff
@@ -33,6 +33,11 @@ jobs:
           pip install flake8 pytest pytest-cov
           if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
           if [ -f tests_requirements.txt ]; then pip install -r tests_requirements.txt; fi
+      - name: Run pre-commit hooks
+        env:
+          LEAKTK_PATTERN_SERVER_AUTH_TOKEN: ${{ secrets.LEAKTK_PATTERN_SERVER_AUTH_TOKEN }}
```
Comment (ebattat) [inline .github/workflows/Build.yml:38]: Why we need to pass token LEAKTK_PATTERN_SERVER_AUTH_TOKEN ?

**Review by ebattat — COMMENTED** [review]
Is it the best GtiHub actions solution for secret leak ?

**Comment by athiruma** [comment]
> Is it the best GtiHub actions solution for secret leak ?

Scan secrets in code, not in GitHub axtions.

**Comment by athiruma** [comment]
> Is it scan all the PR changes ?

Yes

File: .github/workflows/Build.yml (hunk 1 above)
Comment (athiruma) [reply .github/workflows/Build.yml:38]: rh-gitleaks token

**Comment by athiruma** [comment]
It does not need to be added to GitHub Actions. 
Pre-commit will execute before the `git commit` command, ensuring strict checks. After pushing the code to the repo, there is no need for further checking

**Review by ebattat — APPROVED** [review]
/LGTM

**Pull request merged**